### PR TITLE
docs: use PlantUML instead of LaTeX to generate the Schema Isolation image

### PR DESCRIPTION
**Pull request by laurenceisla**
As agreed in https://github.com/PostgREST/postgrest/pull/3384#issuecomment-2043301604

### CODE REVIEW

File: docs/_diagrams/uml/sch-iso.uml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+database "PostgreSQL DB" {
+  rectangle internal <<schema>> {
+    rectangle tables_internal as "tables"
+  }
+  rectangle private <<schema>> {
+    rectangle tables_private as "tables"
+  }
+  rectangle core <<schema>> {
+    rectangle tables_core as "tables"
+  }
```
Comment (steve-chavez) [inline docs/_diagrams/uml/sch-iso.uml:35]: How about just having `public` and `extensions` as schemas.

I recall a user reported internal, core, private was confusing.

File: nix/tools/docs.nix [hunk 2]
```diff
@@ -90,8 +90,8 @@ let
         ${pdflatex}/bin/pdflatex -halt-on-error -output-directory="$tmpdir" db.tex
         ${imagemagick}/bin/convert -density 300 "$tmpdir/db.pdf" ../_static/db.png
```
[the remark this answers is not in the document]
Comment (laurenceisla) [reply nix/tools/docs.nix]: Yes. I removed them in the latest commit.

File: docs/_diagrams/uml/sch-iso.uml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    rectangle vf_api as "views + functions" #f8e71c
+  }
+
+  tables_public <-- vf_api #181818
+  tables_extensions <-- vf_api #181818
+}
+
+vf_api <-[thickness=4]-> () PostgREST #7ed321
+
+@enduml
```
Comment (steve-chavez) [inline docs/_diagrams/uml/sch-iso.uml:43]: I don't think the colors add much here. How about this?

```suggestion
@startuml

database "PostgreSQL" {
  node public {
    rectangle tables_public as "tables"
  }

  node extensions {
  }
  node API {
    rectangle vf_api as "views + functions" #f8e71c
  }

  tables_public <-- vf_api 
  extensions <-- vf_api 
}

vf_api <-[thickness=4]-> () PostgREST 

@enduml
```

**Review by steve-chavez — APPROVED** [review]
Great work! :100:

**Pull request merged**